cmd/controller-manager/app/controller_manager.go

```diff
@@ -111,21 +112,30 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 		rest.AddUserAgent(k8sKubeconfig, controllerManagerAgentName),
 	)
 	if err != nil {
-		glog.Fatalf("Invalid k8s API configuration: %v", err)
+		return fmt.Errorf("invalid Kubernetes API configuration: %v", err)
```

Comment:
TLDR: I wouldn't expect this to fix a bug, I believe it is worth doing for consistency.

> The behavior shouldn't change beyond we actually exit instead of continuing to run.

Sorry if I was unclear in my previous comment.  **`glog.Fatalf` invokes `os.Exit(255)`.**  Do correct me if we have observed or have reason to believe otherwise.

Note: If we have been using glog.Fatal elsewhere and expecting the program to continue running afterward, we should revisit that.

I do agree it is better to be consistent with how we treat fatal errors, so so always returning SGTM vs always exiting via `glog.Fatal` from here.  In the future, we may want to consider having main call `glog.Fatalf` instead of writing directly to stderr (outside the scope of this PR).  This would allow us to configure the logger to write both to stderr and to the container terminationmessage file.

FWIW, the inlined glog docs from [here](https://github.com/golang/glog/blob/master/glog.go#L1147)

```go
// Fatalf logs to the FATAL, ERROR, WARNING, and INFO logs,
// including a stack trace of all running goroutines, then calls os.Exit(255).
// Arguments are handled in the manner of fmt.Printf; a newline is appended if missing.
func Fatalf(format string, args ...interface{}) {

```